Rollbar/RollbarNotifier.m

```diff
@@ -330,12 +342,29 @@ - (BOOL)sendPayload:(NSData*)payload {
     [request setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
     [request setValue:self.configuration.accessToken forHTTPHeaderField:@"X-Rollbar-Access-Token"];
     [request setHTTPBody:payload];
+
+    __block BOOL result = NO;
+    if (IS_IOS7_OR_HIGHER) {
+        // This requires iOS 7.0+
+        dispatch_semaphore_t sem = dispatch_semaphore_create(0);
+        NSURLSessionDataTask *dataTask = [[NSURLSession sharedSession] dataTaskWithRequest:request completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+            result = [self checkPayloadResponse:response error:error data:data];
+            dispatch_semaphore_signal(sem);
+        }];
+        [dataTask resume];
+        dispatch_semaphore_wait(sem, dispatch_time(DISPATCH_TIME_NOW, 5*NSEC_PER_SEC));
```

Comment:
What happens if we timeout here?

Reply:
Changed this to forever. The semaphore should take care of itself when `[[NSURLSession sharedSession] dataTaskWithRequest` times out, which should hit the callback and trigger dispatch_semaphore_signal while takes care of the response/error logging/exception handling etc, just like in previous versions.